src/domains/search/functions/data-mapping.ts

```diff
@@ -49,30 +28,69 @@ export const makeDisplayDataFromPhotonResult = (
   };
 };
 
-export function makeStyles(result: EnrichedSearchResult) {
+export function makeFeatureId(result: EnrichedSearchResult) {
   if (result.photonResult) {
-    return `osm-category-${result.photonResult.properties.osm_key || "unknown"}-${result.photonResult.properties.osm_value || "unknown"}`;
+    return `${result.photonResult.properties.osm_key}:${result.photonResult.properties.osm_id}`;
   }
 
   if (result.placeInfo) {
-    return "ac-place-info";
+    return result.placeInfo.properties._id;
   }
 
-  return undefined;
+  console.warn("No photonResult or placeInfo found in result", result);
+
+  return "unknown-should-not-happen";
 }
 
-export function makeFeatureId(result: EnrichedSearchResult) {
-  if (result.photonResult) {
-    return `${result.photonResult.properties.osm_key}:${result.photonResult.properties.osm_id}`;
+const typeMap = {
+  N: "node",
+  W: "way",
+  R: "relation",
+} as const;
+
+export function mapOsmType(feature: PhotonResultFeature) {
+  return typeMap[feature.properties.osm_type || "N"] || "node";
+}
+
+const collectionMap = {
+  elevator: "elevators",
+  "highway:elevator": "elevators",
+  building: "buildings",
+  "place:house": "entrances_or_exits",
```

Comment:
quip (non-blocking): From my understanding `place:house` is a different entity on OSM than our `entrances_and_exits` table – the nouse numbers are their own nodes so you can put them at a specific position on a map, usually in the center of a building, but we can figure this out later.

Reply:
Yes, let's tackle this later. I only refactored this and didn't add or change anything in the process.